### ISSUE TITLE
Code in demo application should be readonly

### ISSUE THREAD

**Issue by rahulporuri** [enthought/traitsui#1124]
At the moment, the code shown in the ETS Demo application can be edited by the user. I don't think we allow/save user edits to the code so to remove any ambuigity, the code should be readonly.

![code-readonly](https://user-images.githubusercontent.com/1926457/89761775-ac81bd00-dade-11ea-8905-2a7813ace97b.gif)

**Comment by kitchoi** [comment]
I think the idea is that one can edit the code and press the "Run" button to see the new behaviour caused by the edit.
Is that functionality not working?

**Comment by rahulporuri** [comment]
> I think the idea is that one can edit the code and press the "Run" button to see the new behaviour caused by the edit.
Is that functionality not working?

Ahh. That makes sense. It does work but it wasn't obvious that I could update the code and see the results by manually clicking "Run".

**Comment by kitchoi** [comment]
Agreed some way of indicating the run button can be clicked to test the new code would be a good UX enhancement.

Currently the Run button is always enabled, and I think there is value in keeping its enabled even if the code has not changed (e.g. for running the same thing twice).

Maybe a small instruction next to the button or changing the button text from "Run" to "Re-run" would be a good hint.

**Comment by rahulporuri** [comment]
I updated the issue description to reflect the change.

**Comment by kitchoi** [comment]
Thanks!

**Comment by kitchoi** [comment]
@corranwebster  This UX issue can also be fixed by removing the Demo tab so that the function of Run is more obvious.
At the moment, with the demo UI being visible, it is not obvious that one needs to press the Run button to refresh it. 
The awkwardness is more apparent considering an example that opens a popup:

https://user-images.githubusercontent.com/3673984/103523839-6c97e980-4e74-11eb-9f19-25cce7836f5e.mp4

After editing the code, one needs to press two buttons: "Run", and then "Run demo..." for the new code to be used.
It is tempting to think pressing "Run demo..." is enough, but it isn't.

**Comment by corranwebster** [comment]
There is a potential refactor that could resolve some of these issues.  Following #1456 you can now hide the demo tab by having the `demo` variable be `None` (or not set).

This code https://github.com/enthought/traitsui/blob/5a8f5d4f80ccadb0cdd94ccdb28183ca35dbb432/ets-demo/etsdemo/app.py#L474-L482
is responsible for handling the run button click.  We could modify things so that the automatic run during `init` and the "Run" button do different things: in particular, have the run button look for `modal_popup` and `popup` names in the namespace and call `edit_traits()` on those objects directly; while the automatic run on init just ignores that.

**Comment by corranwebster** [comment]
Another possibility is to remove the `run` button completely and have the code automatically re-execute as long as it is syntactically valid, like we do in the Kiva Explorer.
